todo-plainJS.js

```diff
@@ -0,0 +1,154 @@
+let taskList = JSON.parse(localStorage.getItem('taskList')) ?? [];
+
+window.addEventListener('DOMContentLoaded', () => {
+    drawHTML(filterTasks());
+    refreshCount();
+});
+
+form.addEventListener('submit', (event) => {
+    event.preventDefault();
+    const content = task_content.value;
+
+    if (content === '') {
+        console.log('Input is empty');
+    } else {
+        createTask(content);
+        createTaskHTML(content);
+    }
+
+    saveTaskList();
+    refreshCount();
+    task_content.value = '';
+});
+
+for (let item of filters.children) {
+    item.addEventListener('click', () => {
+        switch (item.id) {
+            case 'filter_done':
+                drawHTML(filterTasks('done'));
+                refreshCount(filterTasks('done'));
+                break;
+            case 'filter_undone':
+                drawHTML(filterTasks('undone'));
+                refreshCount();
+                break;
+            default:
+                drawHTML(filterTasks());
+                refreshCount();
+                break;
+        }
+    });
+}
+
+todo.addEventListener('click', (event) => {
+    const index = Array.from(todo.children).indexOf(event.target);
+
+    if (event.target.parentNode === todo) {
+        event.target.classList.toggle('done');
+        console.log('test');
+        markTask(index);
+        saveTaskList();
+        refreshCount();
+    }
+});
+
+todo.addEventListener('click', (event) => {
+    const index = Array.from(todo.children).indexOf(event.target.parentNode);
+    
+    if (event.target.classList.contains('task-delete_button')){
+        event.target.parentElement.remove();
+        deleteTask(index);
+        saveTaskList();
+        refreshCount();
+    }
+});
+
+function drawHTML(taskList) {
+    while (todo.firstChild) {
```

Comment:
жеска

Reply:
Hmm.. I do not like this approach. It would be much better for performance to draw tasks from localstorage once, on onload event. Updating tasks (deletion or creation) should be applied directly. This will make code simplier too